diff --git a/src/celluloid-view.c b/src/celluloid-view.c
index cd27a8d..f9c0a14 100644
--- a/src/celluloid-view.c
+++ b/src/celluloid-view.c
@@ -1682,18 +1682,20 @@ celluloid_view_queue_render(CelluloidView *view)
 	celluloid_video_area_queue_render(area);
 }
 
+/* This function is automatically called before emitting the GtkGLArea::render signal,
+   and doesn’t normally need to be called by application code. */
 void
 celluloid_view_make_gl_context_current(CelluloidView *view)
 {
-	CelluloidMainWindow *wnd =
-		CELLULOID_MAIN_WINDOW(view);
-	CelluloidVideoArea *video_area =
-		celluloid_main_window_get_video_area(wnd);
-	GtkGLArea *gl_area =
-		celluloid_video_area_get_gl_area(video_area);
-
-	gtk_widget_realize(GTK_WIDGET(gl_area));
-	gtk_gl_area_make_current(gl_area);
+	//CelluloidMainWindow *wnd =
+	//	CELLULOID_MAIN_WINDOW(view);
+	//CelluloidVideoArea *video_area =
+	//	celluloid_main_window_get_video_area(wnd);
+	//GtkGLArea *gl_area =
+	//	celluloid_video_area_get_gl_area(video_area);
+
+	//gtk_widget_realize(GTK_WIDGET(gl_area));
+	//gtk_gl_area_make_current(gl_area);
 }
 
 void
